frontend/src/components/Common/Search.tsx

```diff
@@ -48,6 +49,21 @@ type CustomSearchProps<T> = {
   selectCallback: (selected: T) => void;
 };
 
+export function MemberSearch(query: string, member: Member): boolean {
+  const queryLower = query.toLowerCase();
+  return (
+    (member.email && member.email.toLowerCase().startsWith(queryLower)) ||
+    (member.firstName && member.firstName.toLowerCase().startsWith(queryLower)) ||
+    (member.lastName && member.lastName.toLowerCase().startsWith(queryLower)) ||
+    (member.role && member.role.toLowerCase().startsWith(queryLower)) ||
+    ((member.firstName &&
+      member.lastName &&
+      `${member.firstName.toLowerCase()} ${member.lastName.toLowerCase()}`.startsWith(
+        queryLower
+      )) as boolean)
+  );
+}
```

Comment:
I think this can be made into a React component instead of just a function for the match checker for the `CustomSearch` component.

i.e.
```
const MemberSearch = ({query, member}) => {
    return <CustomSearch ... />
}
```

Reply:
<img width="597" alt="image" src="https://user-images.githubusercontent.com/70241445/141172891-17a5b011-7c58-4692-920a-c5abbc277aff.png">

I wasn't sure if I could make it a React component because the similar code is just in the matchChecker piece of the component, but source, resultRenderer, and selectCallback differ in each instance.

Reply:
Oh good catch, maybe rename it to something like `memberMatchChecker` then to better follow naming conventions since it'd just be function